Automattic/MobilePay-Apple
# Embed MobilePayKit as a local Swift Package

## Description

This PR adds the **StoreKit stack** and **embeds MobilePayKit as a local Swift package**. This lays the groundwork for developing the MobilePayKit library.

- Resolves https://github.com/wordpress-mobile/WordPress-iOS/issues/16862
- Resolves https://github.com/wordpress-mobile/WordPress-iOS/issues/16865

https://user-images.githubusercontent.com/6711616/126185663-4e34c2b8-3781-4c25-8f71-981be11ed1f1.mov

## Note

- Not yet implemented: a button on the PurchasableContentDetail view that consumes the purchased consumable
- I may rewrite the view logic in Swift (instead of SwiftUI)

## How to test

1. Go to Edit Scheme > Run > Options > StoreKit Configuration
2. Choose `Configuration.storekit`
3. Run the iOS app
4. Tap on Rocket Fuel to purchase
5. Tap Confirm
6. Tap OK
7. On Xcode debugger menu, click on the Manage StoreKit Transactions menu item (See example below)

<img width="536" alt="Screen Shot 2021-07-19 at 16 29 19" src="https://user-images.githubusercontent.com/6711616/126186328-1bc968f4-8c18-45ad-84f1-b5033fc1cbe7.png">

8. ✅ There should be a record of Rocket Fuel being purchased (See example below)

<img width="747" alt="Screen Shot 2021-07-19 at 16 32 32" src="https://user-images.githubusercontent.com/6711616/126186582-3f76559e-7fe5-45ae-9068-2754883f6351.png">
